app/javascript/react/containers/ParkFormEditContainer.js

```diff
@@ -0,0 +1,179 @@
+    this.handleAddressChange = this.handleAddressChange.bind(this)
+    this.handleCityChange = this.handleCityChange.bind(this)
+    this.handleStateChange = this.handleStateChange.bind(this)
+    this.handleZipChange = this.handleZipChange.bind(this)
+    this.handleDescriptionChange = this.handleDescriptionChange.bind(this)
+    this.handleFormUpdate = this.handleFormUpdate.bind(this)
+  }
+
+    handleNameChange(event) {
+        this.setState({parkName: event.target.value})
+    }
+
+    handleAddressChange(event) {
+        this.setState({parkAddress: event.target.value})
+    }
+
+    handleCityChange(event) {
+        this.setState({parkCity: event.target.value})
+    }
+
+    handleStateChange(event) {
+        this.setState({parkState: event.target.value})
+    }
+
+    handleZipChange(event) {
+      this.setState({parkZip: event.target.value})
+    }
+
+    handleDescriptionChange(event) {
+      this.setState({parkDescription: event.target.value})
+    }
+
+    handleFormClear(event) {
+        this.setState({
+            parkName: '',
+            parkAddress: '',
+            parkCity: '',
+            parkState: '',
+            parkZip: '',
+            parkDescription: '',
+            errors: [],
+            works: ''
+        })
+    }
+
+    handleFormUpdate(event) {
+        event.preventDefault();
+        let formPayload = {
+            name: this.state.parkName,
+            address: this.state.parkAddress,
+            city: this.state.parkCity,
+            state: this.state.parkState,
+            zip: this.state.parkZip,
+            description: this.state.parkDescription        };
+        fetch(`/api/v1/parks/${this.props.params.id}`, {
+            credentials: 'same-origin',
+            method: 'PATCH',
+            body: JSON.stringify(formPayload),
+            headers: { 'Content-Type': 'application/json'}
+        })
+        .then(response => {
+            if(response.ok) {
+                return response;
+            } else {
+                let errorMessage = `${response.status} (${response.statusText})`,
+                    error = new Error(errorMessage);
+                throw(error);
+            }
+        })
+        .then(response => response.json())
+        .then(body => {
+            this.setState({ updatedPark: body.park, error: body.errors })
+            browserHistory.push(`/parks/${this.props.params.id}`)
+        })
+        .catch(error => console.error(`Error in fetch: ${error.message}`))
+    }
+
+
+
+    componentDidMount() {
+
+
+      fetch(`/api/v1/parks/${this.props.params.id}`)
+      .then(response => {
+          if(response.ok) {
+              return response;
+          } else {
+              let errorMessage = `${response.status} (${response.statusText})`,
+                  error = new Error(errorMessage);
+              throw(error);
+          }
+      })
+      .then(response => response.json())
+      .then(body => {
+          this.setState({
```

Comment:
if it's easier. I recommend just maintaining a `park` object in state. that way you can do things like

```javascript
this.setState({park: body.park})
```